bindings/java/src/main/java/org/ucx/jucx/examples/UcxReadBWBenchmarkReceiver.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2019. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+package org.ucx.jucx.examples;
+
+import org.ucx.jucx.UcxCallback;
+import org.ucx.jucx.UcxRequest;
+import org.ucx.jucx.ucp.*;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Stack;
+
+public class UcxReadBWBenchmarkReceiver {
+
+    private static long bytesInGigabit = 1000L * 1000L * 1000L / 8;
+
+    private static double getBandwith(long nanoTimeDelta, long size) {
+        double sizeInGigabits = (double)size / bytesInGigabit;
+        double secondsElapsed = nanoTimeDelta / 1_000_000_000.0;
+        return sizeInGigabits / secondsElapsed;
+    }
+
+    public static void main(String[] args) throws IOException {
+        Map<String, String> argsMap = new HashMap<>();
+        for (String arg: args) {
+            String[] parts = arg.split("=");
+            argsMap.put(parts[0], parts[1]);
+        }
+
+        String serverHost = argsMap.get("s");
+        int serverPort = Integer.parseInt(argsMap.getOrDefault("p", "55443"));
+        int numIterations = Integer.parseInt(argsMap.getOrDefault("n", "5"));
+
+        Stack<Closeable> resources = new Stack<>();
+
+        UcpContext context = new UcpContext(new UcpParams().requestWakeupFeature()
+            .requestRmaFeature().requestTagFeature());
+        resources.push(context);
+        UcpWorker worker = context.newWorker(new UcpWorkerParams());
+        resources.push(worker);
+        UcpListener listener = worker.newListener(
+            new UcpListenerParams().setSockAddr(
+                new InetSocketAddress(serverHost, serverPort)));
+        resources.push(listener);
+
+        ByteBuffer recvBuffer = ByteBuffer.allocateDirect(4096);
+        UcxRequest recvRequest = worker.recvTaggedNonBlocking(recvBuffer, null);
+
+        System.out.println("Waiting for connections ...");
+
+        while (!recvRequest.isCompleted()) {
+            worker.progress();
+        }
+
+        long remoteAddress = recvBuffer.getLong();
+        long remoteSize = recvBuffer.getInt();
+        int remoteKeySize = recvBuffer.getInt();
+        ByteBuffer rkey = ByteBuffer.allocateDirect(remoteKeySize);
+        for (int i = 0; i < remoteKeySize; i++) {
+            rkey.put(recvBuffer.get());
```

Comment:
isn't there a better way to copy byte buffers?

Reply:
There is, but for native buffers need to use the unsafe feature:
https://github.com/apache/spark/blob/master/common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java#L261

For heap buffers it's easier, just need to call:
`ByteBuffer.wrap(buffer.array(), offset, length)`

But native buffer doesn't have `buffer.array()` method.

Don't want to overcomplicate code, this copy doesn't count in transmission time itself.

Reply:
i would add helper function like copyByteBuffer(dst,src) since it's used twice in the code